### DISCUSSION TITLE
Has anyone used this to do a non -pedal midi controller using a esp32s

### DISCUSSION THREAD

**Discussion by fabiendostie** [alf45tar/PedalinoMini#308] [General]
I am in the process of building a "simple"  10 Pots - some momentary buttons and switches and i think this project could be perfect for it especially that it has a lot of the "wireless" features i am looking for.
 I am not really what you would call an expert in coding but i can do some. I am just wondering if anyone has done something siomilar.

thanks

LeFabHouse

**Comment by alf45tar** [comment]
@fabiendostie I didn't understood what do you want to build.

**Reply by fabiendostie** [reply]
its just a midi controller that has 10 pots few buttons and switches.

**Comment by alf45tar** [comment]
PedalinoMini supports max 6 pots or 36 switches or a mix of them.

**Reply by fabiendostie** [reply]
ok not gonna work for me, thanks!

**Comment by mknerr** [comment]
They are cheap enough you could combine two in one project.

**Comment by alf45tar** [comment]
The limit of 6 pots is due to ESP32 max capacity.

The ESP32 has 18 x 12 bits ADC input channels. These are the GPIOs that can be used as ADC and respective channels:

ADC1_CH0 (GPIO 36)
ADC1_CH1 (GPIO 37)
ADC1_CH2 (GPIO 38)
ADC1_CH3 (GPIO 39)
ADC1_CH4 (GPIO 32)
ADC1_CH5 (GPIO 33)
ADC1_CH6 (GPIO 34)
ADC1_CH7 (GPIO 35)

ADC2_CH0 (GPIO 4)
ADC2_CH1 (GPIO 0)
ADC2_CH2 (GPIO 2)
ADC2_CH3 (GPIO 15)
ADC2_CH4 (GPIO 13)
ADC2_CH5 (GPIO 12)
ADC2_CH6 (GPIO 14)
ADC2_CH7 (GPIO 27)
ADC2_CH8 (GPIO 25)
ADC2_CH9 (GPIO 26)

ADC2 pins cannot be used when Wi-Fi is used. Only 8 remains on ADC1 but 37 and 38 are not always available.
Only 6 are available.

Are reported by @mknerr you can build 2 devices.

**Comment by fabiendostie** [comment]
I could multiplex easy!
